libs/ts-rest/nest/src/lib/api.decorator.ts

```diff
@@ -115,18 +118,57 @@ const getMethodDecorator = (appRoute: AppRoute) => {
 
 @Injectable()
 export class ApiRouteInterceptor implements NestInterceptor {
+  private isAppRouteResponse(
+    value: unknown
+  ): value is { status: HTTPStatusCode; body?: any } {
+    return (
+      value != null &&
+      typeof value === 'object' &&
+      'status' in value &&
+      typeof value.status === 'number'
+    );
+  }
+
   intercept(context: ExecutionContext, next: CallHandler): Observable<any> {
     const res: Response = context.switchToHttp().getResponse();
 
+    const appRoute: AppRoute | undefined = Reflect.getMetadata(
+      tsRestAppRouteMetadataKey,
+      context.getHandler()
+    );
+
+    if (!appRoute) {
+      // this will respond with a 500 error without revealing this error message in the response body
+      throw new Error('Make sure your route is decorated with @Api()');
+    }
+
     return next.handle().pipe(
       map((value) => {
-        if (
-          typeof value === 'object' &&
-          typeof value.status === 'number' &&
-          value.body !== undefined
-        ) {
-          res.status(value.status);
-          return value.body;
+        if (this.isAppRouteResponse(value)) {
+          const statusNumber = value.status;
+
+          const isParsingEnabled = Boolean(
+            Reflect.getMetadata(ParseResponsesSymbol, context.getHandler()) ||
+              Reflect.getMetadata(ParseResponsesSymbol, context.getClass())
+          );
+
+          const responseValidation = checkZodSchema(
+            value.body,
+            isParsingEnabled ? appRoute.responses[statusNumber] : undefined
+          );
+
+          if (!responseValidation.success) {
+            const { error } = responseValidation;
+
+            const message = error.errors.map(
+              (error) => `${error.path.join('.')}: ${error.message}`
+            );
```

Comment:
Using the same approach as `@anatine/zod-nestjs` to better formatted validation errors